commands.js

```diff
@@ -286,6 +287,50 @@ var commands = exports.commands = {
 	},
 	makechatroomhelp: ["/makechatroom [roomname] - Creates a new room named [roomname]. Requires: ~"],
 
+	personalroom: 'groupchat',
+	groupchat: function (target, room, user) {
+		// First and foremost, check the number of personal rooms owned by the user.
+		if (user.personalRooms && user.personalRooms >= MAX_PERSONAL_ROOMS) return this.sendReply("You can't own more than " + MAX_PERSONAL_ROOMS + " personal rooms.");
```

Comment:
The way this is written, this is very easy to get around. I'd suggest just using ResouceMonitor, which uses a more DoS-proof method of limiting resource usage.